djconnectwise/sync.py

```diff
@@ -849,13 +854,13 @@ def get_page(self, *args, **kwargs):
 class BoardChildSynchronizer(Synchronizer):
 
     def _assign_field_data(self, instance, json_data):
-        instance.id = json_data['id']
-        instance.name = json_data['name']
+        instance.id = json_data.get('id')
+        instance.name = json_data.get('name')
         try:
-            board_id = json_data['board']['id']
-        except KeyError:
+            board_id = json_data.get('board').get('id')
```

Comment:
You can actually leave this section as `json_data['board']['id']` because it has a specific try/except block handling this case for backwards compatibility.